src/processor/operator/table_scan/base_table_scan.cpp

```diff
@@ -6,6 +6,17 @@ namespace processor {
 void BaseTableScan::initLocalStateInternal(ResultSet* resultSet, ExecutionContext* context) {
     for (auto& dataPos : outVecPositions) {
         auto valueVector = resultSet->getValueVector(dataPos);
+        if (valueVector->dataType.getTypeID() == common::STRUCT) {
```

Comment:
I think this should go into ProcessorTask::populateResultSet

Reply:
moved to `ProcessorTask::populateResultSet`